hoot-services/src/main/java/hoot/services/utils/DbUtils.java

```diff
@@ -162,10 +162,42 @@ public static Long getMapIdByName(String mapName) {
         return createQuery().select(maps.id).from(maps).where(maps.displayName.eq(mapName)).fetchOne();
     }
 
+    public static Long getMapIdByName(String mapName, Long userId) {
+        return createQuery().select(maps.id).from(maps).where(maps.displayName.eq(mapName).and(maps.userId.eq(userId))).fetchOne();
+    }
+
     public static String getDisplayNameById(long mapId) {
         return createQuery().select(maps.displayName).from(maps).where(maps.id.eq(mapId)).fetchOne();
     }
 
+    public static Long getMapIdFromRef(String mapRef, Long userId) {
+        Long mapId;
+        try {
+            mapId = Long.parseLong(mapRef);
+        }
+        catch (NumberFormatException ignored) {
+            mapId = getMapIdByName(mapRef, userId);
+        }
+        if(mapId == null) {
+            throw new IllegalArgumentException(mapRef + " doesn't have a corresponding map ID associated with it!");
+        }
+        return mapId;
+    }
+
+    public static Long getMapIdFromRef(String mapRef) {
```

Comment:
Could potentially do a call like: getMapIdFromRef(mapRef, null) and in the other getMapIdFromRef method check for null to decide which getMapIdByName method to execute